frontend/src/modules/fluenteditor/components/Editor.js

```diff
@@ -86,10 +86,18 @@ export class EditorBase extends React.Component<EditorProps, State> {
             this.props.entity &&
             !this.state.forceSource &&
             this.props.editor.translation !== prevProps.editor.translation &&
-            this.props.editor.changeSource === 'external' &&
+            this.props.editor.changeSource !== 'internal' &&
             typeof(this.props.editor.translation) === 'string'
         ) {
-            this.analyzeFluentMessage(this.props.editor.translation);
+            if (this.props.editor.changeSource === 'machinery') {
+                this.props.updateTranslation(prevProps.editor.translation, true);
+                if (typeof(this.props.editor.translation) === 'string') {
+                    this.props.addTextToEditorTranslation(this.props.editor.translation);
```

Comment:
I don't understand, shouldn't we replace instead of adding text?

Reply:
Good point, we should replace.